src/diffpy/srmise/dataclusters.py

```diff
@@ -82,12 +82,28 @@ def __eq__(self, other):
         )
 
     def clear(self):
-        """Remove all the points from the cluster list,
-        and reset x, y value sequences to empty list. Reset the resolution to 0."""
+        """
+        Clear all data and reset the cluster object to a transient initial state.
+
+        This method performs the following steps:
+        - Remove all the points from the clusters list.
+        - Reset the x and y value sequences to empty NumPy arrays.
+        - Reset dataorder list to empty NumPy arrays.
+        - Reset the resolution to 0.
+        - Reset current_idx to 0.
+        - Reset lastcluster_idx and lastpoint_idx to None.
+        - Reset status to INIT.
+
+        The purpose of this method is to provide a clean state before creating new clustering operations.
```

Comment:
This is good, it is helpful.  I would also have a statement along the lines of "the object is updated in-place and no new instance is returned.", something like that, as this is something that I might like to know when I try and use it.